src/main/webapp/ChapMap.js

```diff
@@ -298,4 +304,175 @@ class ChapMap {
   panTo(coords) {
     this.googleMap_.panTo(coords);
   }
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// RECEIVE MARKERS FROM THE SERVER
+
+  /**
+   * @Private
+   * Fetches all the map's markers from the server and loads them to the map
+   */
+  loadMarkers_() {
+    firebase.auth().currentUser.getIdToken(/* forceRefresh= */ true)
+        .then(idToken => myMap.getMarkers(idToken))
+        .catch(error => {
+          throw "Problem getting markers";
+        });
+  }
+
+  /**
+   * Retrieves markers from the server and adds them to the map
+   * @param {firebase.idToken} idToken the current user's getIdToken
+   */
+  getMarkers(idToken) {
+    fetch(`/map-server?idToken=${idToken}&idRoom=${roomId}`)
+      .then(response => response.json())
+      .then(markers => myMap.handleMarkers_(markers));
+  }
+
+  /**
+   * @Private
+   * Processes a list of markers and adds them to the map
+   * @param {JSON} markers json array of markers that need to be processed
+   */
+  handleMarkers_(markers) {
+    for (const marker in markers) {
+      myMap.handleMarker(markers[marker]);
+    }
+  }
+
+  /**
+   * Modifies or creates a new PermMarker with the given details
+   * @param {JSON} markerJson json containing marker details
```

Comment:
nit: JSON is not a JS type. This would be `Object`.

Reply:
Whoops, I found the same mistake in a few other places as well so they're fixed now.